controllers/serviceaccount_controller.go

```diff
@@ -85,14 +85,10 @@ func AddServiceAccountProviderControllerToManager(ctx *context.ControllerManager
 	}
 
 	return ctrl.NewControllerManagedBy(mgr).For(controlledType).
-		// Watch a ProviderServiceAccount
-		Watches(
```

Comment:
we should still be watching for simple `ProviderServiceAccount` changes

Reply:
The `.For()` adds a default watch not the controlled type which in this case is the `ProviderServiceAccount` object.